src/hpm.c

```diff
@@ -72,25 +78,21 @@
 /* Macro to check for instruction trap */
 #define MCAUSE_ILLEGAL_INST 0x02
 
-/* Return codes */
-#define METAL_HPM_RET_OK 0
-#define METAL_HPM_RET_NOK 1
-
 int metal_hpm_init(struct metal_cpu *gcpu) {
     struct __metal_driver_cpu *cpu = (void *)gcpu;
 
     /* Check if counters are initialized or pointer is NULL */
     if ((gcpu) && (cpu->hpm_count == 0)) {
         metal_hpm_counter n;
-
+        uintptr_t bitmask;
```

Comment:
Can you change uintptr_t to uint64_t? Making it a 'ptr' type suggests that you are dealing with pointers, but these aren't pointers. This will also make the code do the right thing on a 32-bit system, as uint64_t is always 64 bits, but I am guessing on a 32-bit system, uintptr_t may be only 32 bits.

Reply:
Event selection registers width varies as per platform(32/64bit). 
Counter registers are up to 64bit in width(even on 32bit platform).
used uintptr_t from standard defines, which tracks pointer-width (thus effectively register width) on a platform.
Replaced uintptr_t with metal_cpureg_t type which is defined as per cpu register width.

Reply:
Ah, good catch! This looks fine to me.